src/main/java/org/galatea/starter/service/PriceService.java

```diff
@@ -48,15 +54,42 @@ public AlphaPrices getPricesByStock(String stock, String daysToLookBack) {
       size = "full";
     }
 
-
     //Call Alpha Vantage API based on (stock, size)
-    objPrices = pricesclient.getPricesByStock(stock, size);
-    long processEndTime = System.currentTimeMillis();
-    long processTime = processEndTime - processStartTime;
-    log.info("Response from Alpha Vantage for stock: {} and the response size: {}. Processing Time was {}, response object: {}.", stock, size, processTime, objPrices);
+    obj_prices = pricesclient.getPricesByStock(stock, size);
+    log.info("\n Response from Alpha Vantage for... \n stock: " + stock + " "
+        + "\n response size: " + size +
+        "\n Processing time (ms):....... trace-timing-ms" +
+        "\n Response was... \n" + obj_prices);
 
-    return objPrices;
+    return obj_prices;
 
   }
 
-}
\ No newline at end of file
+
+  // Convert Alpha Vantage prices to InternalPrices object
+  public Collection<StockPrices> ConvertPrices(String stock) {
+
+    for (Entry<Date, ResponsePrices> entry : obj_prices.getAvPrices()
```

Comment:
should not use an internal field for this. we should strive to have as little changing internal state as possible. pass the AlphaPrices in as a parameter instead.